File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.cpp
class Solution {
public:
    int goal(vector <int> &nums, int goal) {
        if (goal < 0) return 0;
        int sum = 0, cnt = 0, l = 0;
        for(int r =0; r < nums.size(); r++) {
            sum += nums[r] % 2;
            while(sum > goal) 
                sum -= nums[l++] % 2;
            cnt += r-l+1;
        }
        return cnt;
    }
    int numberOfSubarrays(vector<int>& nums, int k) {
        return goal(nums, k) - goal(nums, k -1 );
    }
};

File: 1370-count-number-of-nice-subarrays/1370-count-number-of-nice-subarrays.py
class Solution:
    def fuck(self, nums: List[int], goal: int) -> int:
        if goal < 0:
            return 0

        cnt, sum, l = 0, 0, 0

        for r in range(len(nums)):
            sum += nums[r] % 2
            while sum > goal:
                sum -= nums[l] % 2
                l += 1
            cnt += r-l+1

        return cnt 

    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        cnt = self.fuck(nums, k) - self.fuck(nums, k-1)
        return cnt